protocol_connect.go

```diff
@@ -919,9 +922,12 @@ func newConnectWireError(err error) *connectWireError {
 }
 
 func (e *connectWireError) asError() *Error {
-	if e == nil || e.Code == 0 {
+	if e == nil {
 		return nil
 	}
+	if e.Code == 0 {
```

Comment:
Does this need to map other invalid codes to unknown besides zero?

Reply:
Yes. We should map anything outside the known codes to `CodeUnknown`.